libraries/shared/storage/utils/decoder.go

```diff
@@ -51,3 +60,52 @@ func decodeUint48(raw []byte) string {
 func decodeAddress(raw []byte) string {
 	return common.BytesToAddress(raw).Hex()
 }
+
+func decodePackedSlot(raw []byte, packedTypes map[int]ValueType) map[int]string {
+	storageSlotData := raw
+	decodedStorageSlotItems := map[int]string{}
+	numberOfTypes := len(packedTypes)
+
+	for position := 0; position < numberOfTypes; position++ {
```

Comment:
could maybe replace this with `for k, v := range packedTypes`

Reply:
Maps also don't have a guaranteed iteration order (golang actually purposefully randomizes the intersection order 😆: https://blog.golang.org/go-maps-in-action), so if we iterate over them with `for k, v := range packedTypes` they won't necessarily decode the raw data in the right order.

I think an alternative to the current implementation could be to sort the `packedTypes` keys, and then do something like this
```
var keys []int
for k := range packedTypes {
    keys = append(keys, k)
}
sort.Ints(keys)
for _, k := range keys {
   // similar loop
}
```
Do you think that's more readable?

Reply:
Good point! I'm fine to stick with the current implementation given that we need a guaranteed iteration order.

I am wondering how much it would muck things up to try an implementation that doesn't require a specific iteration order. Thinking that perhaps we could implement a function that returns the relevant slice for a given index provided the indexes and types of all values packed into the slot, but I'm also not sure whether or not that would be more readable 🤔

Reply:
yeah, i need to think about this a bit more. 🤔 do you think it's worth adding a story in the back log to address this later?